src/plugins/recma-mdx-frontmatter.mjs

```diff
@@ -0,0 +1,49 @@
+/// <reference path="./plugins.d.ts" />
+import { fromJs } from 'esast-util-from-js';
+import { visit } from 'estree-util-visit';
+
+/**
+ * Plugin to pass frontmatter to getStaticProps and into MDXProvider's `wrapper`
+ * @returns {import('unified').Plugin<[RecmaMdxFrontmatterPluginOptions], import('estree').Program>}
+ */
+const recmaMdxFrontmatter = (options) => {
+  const { name = 'frontmatter' } = options || {};
+  /**
+   * Transformer
+   * @param {import('estree').Program} tree
+   */
+  const transformer = (tree) => {
+    const frontmatter = {};
+    visit(tree, (node) => {
+      // look for `frontmatter` variable created by remark-mdx-frontmatter
+      if (node.type === 'VariableDeclarator' && node.id.name === name) {
+        if (!Array.isArray(node.init.properties)) return;
```

Comment:
note: should also check for a valid length before attempting to access props on `undefined` (if it is an array but is empty)